okhttp/src/main/java/okhttp3/Address.kt

```diff
@@ -184,17 +184,17 @@ class Address(
     return result
   }
 
-  internal fun equalsNonHost(that: Address): Boolean {
-    return this.dns == that.dns &&
-        this.proxyAuthenticator == that.proxyAuthenticator &&
-        this.protocols == that.protocols &&
-        this.connectionSpecs == that.connectionSpecs &&
-        this.proxySelector == that.proxySelector &&
-        this.proxy == that.proxy &&
-        this.sslSocketFactory == that.sslSocketFactory &&
-        this.hostnameVerifier == that.hostnameVerifier &&
-        this.certificatePinner == that.certificatePinner &&
-        this.url.port == that.url.port
+  internal fun equalsNonHost(that: Address): Boolean = with(this) {
```

Comment:
Do we need `with` here? Can simply get rid of `this`, although I'd keep it, it reads better

Reply:
I have a hard time justifying the use of `with` anywhere, but as you state `this` is already in scope so it's pointless.

Reply:
Well..Reverted.